cns/requestcontroller/kubecontroller/crdrequestcontroller.go

```diff
@@ -0,0 +1,168 @@
+package kubecontroller
+
+import (
+	"context"
+	"errors"
+	"os"
+
+	"github.com/Azure/azure-container-networking/cns/logger"
+	"github.com/Azure/azure-container-networking/cns/restserver"
+	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
+	"k8s.io/apimachinery/pkg/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const nodeNameEnvVar = "NODENAME"
+const k8sNamespace = "kube-system"
+const prometheusAddress = "0"
+
+// crdRequestController
+// - watches CRD status changes
+// - updates CRD spec
+type crdRequestController struct {
+	mgr        manager.Manager //Manager starts the reconcile loop which watches for crd status changes
+	APIClient  APIClient       //Relying on the APIClient interface to more easily test
+	nodeName   string          //name of node running this program
+	Reconciler *CrdReconciler
+}
+
+// GetKubeConfig precedence
+// * --kubeconfig flag pointing at a file at this cmd line
+// * KUBECONFIG environment variable pointing at a file
+// * In-cluster config if running in cluster
+// * $HOME/.kube/config if exists
+func GetKubeConfig() (*rest.Config, error) {
+	k8sconfig, err := ctrl.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+	return k8sconfig, nil
+}
+
+//NewCrdRequestController given a reference to CNS's HTTPRestService state, returns a crdRequestController struct
+func NewCrdRequestController(restService *restserver.HTTPRestService, kubeconfig *rest.Config) (*crdRequestController, error) {
+
+	//Check that logger package has been intialized
+	if logger.Log == nil {
+		return nil, errors.New("Must initialize logger before calling")
```

Comment:
Should we fallback to logging to stdout if the logger isn't initialized? Maybe not, but wanted to ask in case

Reply:
Hmm that's a good question, I would say, in order to capture the logs in AI telemetry and stuff then no